src/dune_cache_storage/layout.ml

```diff
@@ -1,25 +1,29 @@
 open Stdune
 open Import
 
-let default_root_dir =
-  lazy
-    (let cache_dir = Xdg.cache_dir (Lazy.force Dune_util.xdg) in
-     Path.L.relative (Path.of_filename_relative_to_initial_cwd cache_dir) [ "dune"; "db" ])
-;;
+let ( / ) = Path.relative
 
-let root_dir =
+(** The directory containing the build cache.
+    Uses [$DUNE_CACHE_ROOT] if set, or
+    [$DUNE_CACHE_HOME/db] if set, or
+    [Dune_util.default_cache_dir/db] otherwise. *)
+let build_cache_dir =
   lazy
-    (let var = "DUNE_CACHE_ROOT" in
-     match Sys.getenv_opt var with
-     | None -> Lazy.force default_root_dir
+    (let high_priority_var = "DUNE_CACHE_ROOT" in
+     match Sys.getenv_opt high_priority_var with
      | Some path ->
-       if Filename.is_relative path
-       then failwith (sprintf "%s should be an absolute path, but is %s" var path);
-       Path.of_filename_relative_to_initial_cwd path)
+       Dune_util.check_absolute ~var:high_priority_var ~path;
+       Path.external_ (Path.External.of_string path)
+     | None ->
+       let low_priority_var = "DUNE_CACHE_HOME" in
```

Comment:
I don't think we should be reading from `DUNE_CACHE_HOME` from the env in two separate places. This should use the `Dune_util.cache_home_dir`.